modules/jenkins.test.js

```diff
@@ -0,0 +1,111 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const jenkins = require("./jenkins");
+const fs = require("fs");
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Tests
+// -----------------------------------------------------------------------------------------
+
+describe("jenkins", () => {
+    describe("configureCredentials", () => {
+        it("when url is empty then returns false", () => {
+            // Arrange
+            const url = "";
+            const username = "aaa";
+            const token = "aaa";
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+            // Assert
+            expect(result).toBe(false);
+        });
+
+        it("when username is empty then returns false", () => {
+            // Arrange
+            const url = "aaa";
+            const username = "";
+            const token = "aaa";
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+            // Assert
+            expect(result).toBe(false);
+        });
+
+        it("when token is empty then returns with errors", () => {
+            // Arrange
+            const url = "aaa";
+            const username = "aaa";
+            const token = "";
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+            // Assert
+            expect(result).toBe(false);
+        });
+
+        it("when url, username and token provided then returns true ", () => {
+            // Arrange
+            jenkins.writeToConfig = jest.fn(() => true);
+            const url = "aaa";
+            const username = "aaa";
+            const token = "aaa";
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+            // Assert
+            expect(result).toBe(true);
+        });
+    });
+
+    describe("getConfig", () => {
+        it("when config does not exist, then returns undefined", () => {
+            // Arrange
+            jest.mock("fs");
+            fs.readFileSync = jest.fn(() => undefined);
+
+            // Act
+            const result = jenkins.getConfig();
+
+            // Assert
+            expect(result).toBeUndefined();
+        });
+
+        it("should return a JSON config object", () => {
+            // Arrange
+            jest.mock("fs");
```

Comment:
Do we need to mock the `fs` module this way? Not 100% sure myself, since it's a core node module. I've only needed to call `jest.mock()` once at the top of a test file for the behavior to be replaced. If we can get away with doing a `jest.spyOn(fs, "readFileSync").mockImplementation(...)`, I'd prefer to see these arrangements swapped